framework/PostProcessors/FTgenerator.py

```diff
@@ -0,0 +1,288 @@
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import utils, xmlUtils
+from utils import InputData, InputTypes
+from sympy.logic import SOPform, POSform
+from itertools import product
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class FTgenerator(PostProcessor):
+  """
+    This postprocessor is designed to create a fault tree in OpenPSA format from a set of simulation data
+    The postprocessor acts only on PointSet and return a reduced PointSet and print of file the FT in OpenPSA format
+  """
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.inputVars = None    # variable associated with the lower limit of the value dimension
+    self.topEventID = None   # variable associated with the upper limit of the cost dimension
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(FTgenerator, cls).getInputSpecification()
+
+    inputSpecification.addSub(InputData.parameterInputFactory("topEventID", contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("inputVars" , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("fileName"  , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("simOnly"   , contentType=InputTypes.BoolType))
+
+    typeAllowedFormats = InputTypes.makeEnumType("calculationFormat", "calculationFormatType", ["sop","pos"])
+    inputSpecification.addSub(InputData.parameterInputFactory("type" , contentType=typeAllowedFormats))
+
+    return inputSpecification
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    topEventID  = paramInput.findFirst('topEventID')
+    self.topEventID = topEventID.value.strip()
+
+    inputVars = paramInput.findFirst('inputVars')
+    self.inputVars = inputVars.value.split(",")
+    for var in self.inputVars:
+      var = var.strip()
+
+    fileName  = paramInput.findFirst('fileName')
+    self.fileName = fileName.value
+
+    simOnly  = paramInput.findFirst('simOnly')
+    self.simOnly = simOnly.value
+
+    typeConv = paramInput.findFirst('type')
+    self.type = typeConv.value.lower()
```

Comment:
maybe worth to add lower option in InputTypes, and a discussion with our teams. No need to change here.